src/schema-generation/utils/flex-search-utils.ts

```diff
@@ -37,3 +41,19 @@ export function orderArgMatchesPrimarySort(
 
     return true;
 }
+
+/**
+ * Generates a list of orderBy enum values that exactly represent the primary search
+ */
+export function getSortClausesForPrimarySort(
+    objectType: RootEntityType,
+    orderByType: OrderByEnumType,
+): ReadonlyArray<OrderByEnumValue> {
+    // this would be cleaner if the primary sort was actually parsed into a ModelComponent (see e.g. the Index and IndexField classes)
+    return objectType.flexSearchPrimarySort.map((clause) =>
+        orderByType.getValueOrThrow(
+            clause.field.path.replace('.', '_') +
+                (clause.direction === OrderDirection.ASCENDING ? '_ASC' : '_DESC'),
+        ),
+    );
+}
```

Comment:
Not really related to this MR since you just extracted the logic into a function. Wouldn't `clause.field.path.replace(...)` fail when the field in the clause has more than two segments (e.g. nested value objects) since it only replaces the first occurrence?

Reply:
hm yes you're right. I guess we never had primary sorts with more than two segments